### DISCUSSION TITLE
How to focus map to multiple markers with minimum zoom level change (when it's required if with pan all elements are not able to focus in current viewport) [currently using fitbounds].

### DISCUSSION THREAD

**Discussion by vissharm** [visgl/deck.gl#6408] [General]
User should be able to bring focus to multiple selected markers with minimum zoom level change. i.e.  only pan operation but allow minimum zoom change if all selected items are not coming into view port.

Current viewport fitbound implementation changes the zoom level as it try to completely fit to new bounds passed.

I think my solution would require two steps, how to do that... will see

1. First check get current selected items (selection not from  deckgl canvas but from other ui layout) center latlng. Are all the items already within bounds. Don't do anything. If not, go to further step.
2. If we update the map to "selected items" center lat lng, then Are all the items within new bounds ? - Need to calculate center lat/lng of selected items 
3. If yes, then no need to do anything and it seems like only pan operation happened without zoom level change - Deckgl inbuilt method to do that ? 
4. If no, then how could we find just minimum zoom level to bring the items just in focus and let the visibility of other items if possible.

**Comment by Pessimistress** [comment]
I believe `fitBounds` takes a `maxZoom` option.

**Reply by vissharm** [reply]
fitbounds do both pan and zoom even when zoom it is not required. @Pessimistress

**Reply by Pessimistress** [reply]
Set `maxZoom` to the current zoom.

**Reply by vissharm** [reply]
But if with that zoom currently selected items from "grid/or some other section" not getting inside bounds then ? 
@Pessimistress you might be not looking at my problem statement as a whole.

**Comment by vissharm** [comment]
@Pessimistress  I updated the discussion description again. Have a look at steps once again.
